### ISSUE TITLE
enum param not supported?

### ISSUE THREAD

**Issue by Dowwie** [nox/serde_urlencoded#35]
The following example features an enum parameter.  Such an Enum would constrain acceptable values passed for a parameter within the query string and so would be a valuable addition to serde_urlencoded.  Currently, however, serde_urlencoded raises the following error: `Err` value: Error { err: "invalid type: string 
\"brown\", expected enum HairColor" }',

I've also tried declaring a serde tag attribute on the enum, #[serde(untagged)], but that didn't help.

Am I doing this wrong?

```rust
extern crate serde_urlencoded;
#[macro_use] extern crate serde_derive;
extern crate serde;


#[derive(Deserialize)]
enum HairColor {
    brown,
    black,
}


#[derive(Deserialize)]
struct MyParams {
    name: String,
    age: u32,
    hair_color: HairColor
}


fn main() {
    let params = "name=Darin&age=38&hair_color=brown";
    let decoded = serde_urlencoded::from_str::<MyParams>(&params).unwrap();
}

```

**Comment by mqudsi** [comment]
Are enums not covered by the tests?

**Comment by hayd** [comment]
Is this implemented by #30 (not merged) ?

**Comment by Dowwie** [comment]
@hayd Yes, it appears that the PR *should* cover this request!  Does anyone know why it hasn't been approved?
